core/src/main/java/com/linecorp/armeria/common/http/HttpResponse.java

```diff
@@ -36,6 +39,35 @@ static HttpResponse of(Publisher<? extends HttpObject> publisher) {
         return new PublisherBasedHttpResponse(publisher);
     }
 
+    /**
+     * Creates a new failed instance.
+     */
+    static HttpResponse ofFailed(Throwable cause) {
+        final DefaultHttpResponse res = new DefaultHttpResponse();
+        res.close(cause);
+        return res;
+    }
+
+    /**
+     * Creates a new instance that delegates to the {@link HttpResponse} produced by the specified
+     * {@link CompletionStage}. If the specified {@link CompletionStage} fails, the returned response will be
+     * closed with the same cause as well.
+     */
+    static HttpResponse from(CompletionStage<? extends HttpResponse> stage) {
+        requireNonNull(stage, "stage");
+        final DeferredHttpResponse res = new DeferredHttpResponse();
+        stage.whenComplete((delegate, cause) -> {
+            if (cause != null) {
+                res.close(cause);
+            } else if (delegate == null) {
+                res.close(new NullPointerException("stage produced a null delegate: " + stage));
```

Comment:
```delegate stage produced a null response``` might be clearer. Alternatively, since this seems so rare, just leaving it to the ```requireNonNull``` in ```DeferredHttpResponse#delegate``` is probably enough too.